ArmEmulator/ArithmeticTests.fs

```diff
@@ -454,17 +454,19 @@ module ArithmeticTests
                 }
 
                 // Use my code with the same input values to obtain hopefully the same output
-                let results = doArithmetic parsed cpuData
+                let results = execArithmeticInstr parsed cpuData
 
-                let resCpu = results
-                let localRegs = 
-                    resCpu.Regs
-                    |> Map.toList
-                    |> List.map (fun (_, x) -> x)
+                match results with
+                | Ok resCpu ->
+                    let localRegs = 
+                        resCpu.Regs
+                        |> Map.toList
+                        |> List.map (fun (_, x) -> x)
 
-                // Expecto checks
-                Expect.equal regsActual.[..regsActual.Length - 2] localRegs.[..localRegs.Length - 2] "Registers"
-                Expect.equal flagsActual resCpu.Fl "Flags"
+                    // Expecto checks
+                    Expect.equal regsActual.[..regsActual.Length - 2] localRegs.[..localRegs.Length - 2] "Registers"
+                    Expect.equal flagsActual resCpu.Fl "Flags"
+                | Error _ -> ()
```

Comment:
Check the errors too!

Reply:
What do you mean by check the error? Print it out?

Reply:
Like, if there should be an error, then check that it gives the correct error message. 
You'd need a match statement like
```fsharp
match opcode, opnum with
RSB, x when x < 0 -> Error "Invalid immediate operand value"
...```

Or something like that. You can do that in another PR if it's too much work - it's not necessary for what I need right now.

Reply:
Ah ok.  It's not really practical for RSB and RSC since the value of op2 is only evaluated at runtime.  It's a tricky error to test and visUAL also only detects the error at runtime.  So it would not be easy or worth it in my opinion for testing those error messages atm.  Of course if we had more time it would be worth testing